BehatCommands.php

```diff
@@ -0,0 +1,202 @@
+     */
+    function opCreateNode($node)
+    {
+        // Default status to 1 if not set.
+        if (!isset($node->status)) {
+            $node->status = 1;
+        }
+        // If 'author' is set, remap it to 'uid'.
+        if (isset($node->author)) {
+            $user = user_load_by_name($node->author);
+            if ($user) {
+                $node->uid = $user->id();
+            }
+        }
+
+        // Attempt to decipher any fields that may be specified.
+        $this->expandEntityFields('node', $node);
+
+        $entity = entity_create('node', (array) $node);
+        $entity->save();
+
+        $node->nid = $entity->id();
+
+        return (array) $node;
+    }
+
+    /**
+     * Delete a node.
+     */
+    function opDeleteNode($node) {
+        $node = $node instanceof NodeInterface ? $node : Node::load($node->nid);
+        if ($node instanceof NodeInterface) {
+           $node->delete();
+        }
+    }
+
+    /**
+     * Create a taxonomy term.
+     */
+    function opCreateTerm($term) {
+        $term->vid = $term->vocabulary_machine_name;
+
+        // Attempt to decipher any fields that may be specified.
+        $this->expandEntityFields('taxonomy_term', $term);
+
+        $entity = entity_create('taxonomy_term', (array)$term);
+        $entity->save();
+
+        $term->tid = $entity->id();
+        return $term;
+    }
+
+    /**
+     * Delete a taxonomy term.
+     */
+    function opDeleteTerm(\stdClass $term) {
+        $term = $term instanceof TermInterface ? $term : Term::load($term->tid);
+        if ($term instanceof TermInterface) {
+            $term->delete();
+        }
+    }
+
+    /**
+     * Check if this is a field.
+     */
+    function opIsField($is_field_info) {
+        list($entity_type, $field_name) = $is_field_info;
+        return $this->isField($entity_type, $field_name);
+    }
+
+    /**
+     * Get all of the field attached to the specified entity type.
+     *
+     * @see Drupal\Driver\Cores\Drupal8\getEntityFieldTypes in Behat
+     */
+    function getEntityFieldTypes($entity_type) {
+        $return = array();
+        $fields = \Drupal::entityManager()->getFieldStorageDefinitions($entity_type);
+        foreach ($fields as $field_name => $field) {
+            if ($this->isField($entity_type, $field_name)) {
+                $return[$field_name] = $field->getType();
+            }
+        }
+        return $return;
+    }
+
+    function isField($entity_type, $field_name) {
+        $fields = \Drupal::entityManager()->getFieldStorageDefinitions($entity_type);
+        return (isset($fields[$field_name]) && $fields[$field_name] instanceof FieldStorageConfig);
+    }
+
+    /**
+     * Expands properties on the given entity object to the expected structure.
+     *
+     * @param \stdClass $entity
+     *   Entity object.
+     *
+     * @see Drupal\Driver\Cores\AbstractCore\expandEntityFields
+     */
+    function expandEntityFields($entity_type, \stdClass $entity) {
```

Comment:
This and the other methods that don't already should probably specify visibility (`protected`, `public`, etc) for clarity.